src/Compilation.zig

```diff
@@ -4028,7 +4028,14 @@ fn updateCObject(comp: *Compilation, c_object: *CObject, c_obj_prog_node: *std.P
                 args_dir, args_hex_hash[0..],
             });
             try output_dir.handle.writeFile(args_file, args);
-            const args_file_path = try output_dir.handle.realpathAlloc(allocator, args_file);
+
+            const args_file_path = args_file_path: {
+                if (output_dir.path) |odp| {
+                    break :args_file_path try fs.path.join(allocator, &[_][]const u8{ odp, args_file });
+                } else {
+                    break :args_file_path args_file;
+                }
+            };
```

Comment:
As of 9bcd48e40e272408879a442b21099ea5ab6b85b4 there should be no more uses of realpath in the compiler. I believe that means this commit can be removed from the changeset.